modules/core/src/v2/coins/eth2.ts

```diff
@@ -0,0 +1,767 @@
+  recipient: Recipient;
+}
+
+export interface RecoverOptions {
+  userKey: string;
+  backupKey: string;
+  walletPassphrase?: string;
+  walletContractAddress: string;
+  recoveryDestination: string;
+}
+
+export interface RecoveryInfo {
+  id: string;
+  tx: string;
+  backupKey?: string;
+  coin?: string;
+}
+
+interface GetSendMethodArgsOptions {
+  recipient: Recipient;
+  expireTime: number;
+  contractSequenceId: number;
+  signature: string;
+}
+
+interface SendMethodArgs {
+  name: string;
+  type: string;
+  value: any;
+}
+
+interface FeeEstimate {
+  gasLimitEstimate: number;
+  feeEstimate: number;
+}
+
+export class Eth2 extends BaseCoin {
+  static createInstance(bitgo: BitGo): BaseCoin {
+    return new Eth2(bitgo);
+  }
+
+  /**
+   * Returns the factor between the base unit and its smallest subdivison
+   * @return {number}
+   */
+  getBaseFactor(): string {
+    // 10^18
+    return '1000000000000000000';
+  }
+
+  getChain(): string {
+    return 'eth2';
+  }
+
+  getFamily(): string {
+    return 'eth2';
+  }
+
+  getFullName(): string {
+    return 'Ethereum 2.0';
+  }
+
+  /**
+   * Flag for sending value of 0
+   * @returns {boolean} True if okay to send 0 value, false otherwise
+   */
+  valuelessTransferAllowed() {
+    return true;
+  }
+
+  /**
+   * Flag for sending data along with transactions
+   * @returns {boolean} True if okay to send tx data (ETH), false otherwise
+   */
+  transactionDataAllowed() {
+    return true;
+  }
+
+  /**
+   * Evaluates whether a address string is valid for this coin
+   * @param address
+   */
+  isValidAddress(address: string): boolean {
+    return this.isValidPub(address);
+  }
+
+  /**
+   * Return boolean indicating whether input is valid public key for the coin.
+   *
+   * @param {String} pub the pub to be checked
+   * @returns {Boolean} is it valid?
+   */
+  isValidPub(pub: string): boolean {
+    try {
+      BLS.PublicKey.fromHex(pub);
+      return true;
+    } catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   * Default gas price from platform
+   * @returns {BigNumber}
+   */
+  getRecoveryGasPrice(): any {
+    return new optionalDeps.ethUtil.BN('20000000000');
+  }
+
+  /**
+   * Default gas limit from platform
+   * @returns {BigNumber}
+   */
+  getRecoveryGasLimit(): any {
+    return new optionalDeps.ethUtil.BN('500000');
+  }
+
+  /**
+   * Default expire time for a contract call (1 week)
+   * @returns {number} Time in seconds
+   */
+  getDefaultExpireTime(): number {
+    return Math.floor(new Date().getTime() / 1000) + 60 * 60 * 24 * 7;
+  }
+
+  /**
+   * Query Etherscan for the balance of an address
+   * @param address {String} the ETH address
+   * @param callback
+   * @returns {BigNumber} address balance
+   */
+  queryAddressBalance(address: string, callback?: NodeCallback<any>): Bluebird<any> {
+    const self = this;
+    return co(function*() {
+      const result = yield self.recoveryBlockchainExplorerQuery({
```

Comment:
Does this work for ETH2 chain?

Reply:
This should - for ETH1 we query etherscan - for ETH2 I set it up to query beaconscan (same API as Etherscan, made by same creators). If there is a better API certainly let me know (this might fail until transferring is enabled in ETH2 though)